tests/api/test_account.py

```diff
@@ -660,16 +661,107 @@ def test_customer_create(
     assert not data["user"]["isStaff"]
     assert data["user"]["isActive"]
 
-    assert send_set_password_customer_email_mock.call_count == 1
-    args, kwargs = send_set_password_customer_email_mock.call_args
-    call_pk = args[0]
-    assert call_pk == new_customer.pk
+    _send_set_password_email_mock.assert_called_once_with(
+        new_customer.email, ANY, "dashboard/customer/set_password"
+    )
 
     customer_creation_event = account_events.CustomerEvent.objects.get()
     assert customer_creation_event.type == account_events.CustomerEvents.ACCOUNT_CREATED
     assert customer_creation_event.user == new_customer
 
 
+@patch("saleor.dashboard.emails._send_set_user_password_email_with_url.delay")
+def test_customer_create_send_password_with_url(
+    _send_set_user_password_email_with_url_mock,
+    staff_api_client,
+    permission_manage_users,
+):
+    email = "api_user@example.com"
+    variables = {
+        "email": email,
+        "send_mail": True,
+        "redirect_url": "https://www.example.com",
+    }
+
+    response = staff_api_client.post_graphql(
+        CUSTOMER_CREATE_MUTATION, variables, permissions=[permission_manage_users]
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["customerCreate"]
+    assert not data["errors"]
+
+    User = get_user_model()
+    new_customer = User.objects.get(email=email)
+    assert new_customer
+
+    token = default_token_generator.make_token(new_customer)
+    _send_set_user_password_email_with_url_mock.assert_called_once_with(
+        new_customer.email, ANY, token, "dashboard/customer/set_password"
+    )
+
+
+def test_customer_create_without_send_password(
+    staff_api_client, permission_manage_users
+):
+    email = "api_user@example.com"
+    variables = {"email": email}
+    response = staff_api_client.post_graphql(
+        CUSTOMER_CREATE_MUTATION, variables, permissions=[permission_manage_users]
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["customerCreate"]
+    assert not data["errors"]
+    staff_user = User.objects.get(email=email)
+    assert staff_user
+
+
+def test_customer_create_without_redirect_url(
+    staff_api_client, permission_manage_users
+):
+    email = "api_user@example.com"
+    variables = {"email": email, "send_mail": True}
+    response = staff_api_client.post_graphql(
+        CUSTOMER_CREATE_MUTATION, variables, permissions=[permission_manage_users]
+    )
+    content = get_graphql_content(response)
+    data = content["data"]["customerCreate"]
+    assert data["errors"][0]["field"] == "redirectUrl"
```

Comment:
Comparing the message as well would be better to ensure that's actually the error we were expecting, and to allow developers to search for the associated test

Reply:
I hope that today we'll merge #4676 which would allow us to check the error codes in cases like this. Relying on error messages comparisons in tests always feels a little bit clunky to me ;)